packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -971,11 +971,33 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 /// A [Navigator] observer that notifies [RouteAware]s of changes to the
 /// state of their [Route].
 ///
+/// [RouteObserver] informs subscribers whenever a route of type `T` is pushed
+/// on top of their own route of type `T` or popped from it. This is for example
+/// useful to keep track of page transitions, e.i. a `RouteObserver<PageRoute>`
+/// will inform subscribed [RouteAware]s whenever the user navigates away from
+/// the current page route to another page route.
```

Comment:
For analytics, you mostly care about transition from one full-screen page to another (at least for this use case). You don't care about transient routes (e.g. an error dialog informing you that your phone number is required to have 10 digits should not show up as a page transition in analytics). Those transient route-specific sub-routes should be popped off the navigation stack before moving to the next page route (and from my experience that's how almost all apps behave).

That's how leafy wants Google Analytics to be wired up and that also made the most sense to me when wiring up Firebase Analytics.

If you really want to be informed about all route changes, you're free to instantiate `RouteObserver<Route>` as mentioned in the docs in the paragraph below.